src/python/WMCore/MicroService/Unified/MSRuleCleaner.py

```diff
@@ -38,10 +158,65 @@ def __init__(self, msConfig, logger=None):
+        self.msConfig.setdefault("runMode", "dryRun")
+        self.mode = self.msConfig["runMode"]
+        self.isRealMode = True if self.mode == "real" else False
         self.emailAlert = EmailAlert(self.msConfig)
 
+        # Building all the Pipelines:
+        # NOTE: The pipeline names need to overlap with the key names in the
+        #       the 'CleanupStatus' the 'RulesToClean' fields in MSRuleCleanerWflow
+        #       because that's how every pipeline finds its relevant flags and
+        #       counters to populate.
+        # TODO: To automate the creation of those lists of flags and counters
+        #       in the workflow describtion so that to avoid any future mistakes
+        #       The proper place for that to happen is self.setPlineMarker()
+        pName = 'plineMSTrCont'
+        self.plineMSTrCont = Pipeline(name=pName, funcLine=[])
+
+        pName = 'plineMSTrBlock'
+        self.plineMSTrBlock = Pipeline(name=pName, funcLine=[])
+
+        pName = 'plineAgentCont'
+        self.plineAgentCont = Pipeline(name=pName,
+                                       funcLine=[Functor(self.setPlineMarker, pName),
+                                                 Functor(self.getMSOutputTransferInfo),
+                                                 Functor(self.getRucioRules, 'container', self.msConfig['rucioWmaAccount']),
+                                                 Functor(self.cleanRucioRules)])
+        pName = 'plineAgentBlock'
+        self.plineAgentBlock = Pipeline(name=pName,
+                                        funcLine=[Functor(self.setPlineMarker, pName),
+                                                  Functor(self.getMSOutputTransferInfo),
+                                                  Functor(self.getRucioRules, 'block', self.msConfig['rucioWmaAccount']),
+                                                  Functor(self.cleanRucioRules)])
+        pName = 'plineArchive'
+        self.plineArchive = Pipeline(name="plineArchive",
+                                     funcLine=[Functor(self.setPlineMarker, pName),
+                                               Functor(self.setClean),
+                                               Functor(self.archive)])
+
+        # Building the different set of plines we will need later:
+        # NOTE: The following are all the functional pipelines which are supposed to include
+        #       a cleanup function and report cleanup status in the MSRuleCleanerWflow object
+        self.cleanuplines = [self.plineMSTrCont,
+                             self.plineMSTrBlock,
+                             self.plineAgentCont,
+                             self.plineAgentBlock]
+        self.agentlines = [self.plineAgentCont,
+                           self.plineAgentBlock]
+        self.mstrlines = [self.plineMSTrCont,
+                          self.plineMSTrBlock]
+
+        # Zero the 'cleaned' and 'archived' counters:
```

Comment:
I do not see any need for a separate method, because these counters are initialized only once during the thread creation and never reset again until the end. I would rather change the comment saying:
`Initializing the 'cleaned' and 'archived' counters`